### DISCUSSION TITLE
Wacom Tablet: Plugin to reduce the drawing area to part of a screen

### DISCUSSION THREAD

**Discussion by TimoJoFischer** [xournalpp/xournalpp#3032] [Show and tell]
I take notes with xournalpp (i love it, great work :heart:).

Only having one problem with it: To take notes with my small wacom tablet and a large screen i need always scroll in and scroll out -> i wrote my own plugin to fix this issue. i hope it can help aswell some other peoples

What does the plugin do:
- it maps the drawing tablet only to a small display area of my screen (to see the drawingarea: i plot a rectangle with cairo around it)
- with Alt+W/A/S/D the display area is moved
- with Alt+R it maps the displayarea to half of the screen

Example of the shown display area:
![Screenshot_20210413_181039](https://user-images.githubusercontent.com/48179868/114585158-a3f01080-9c83-11eb-8e0c-401ff2e5c727.png)

I don't make a pull request, because:
- only working for linux with cairo, xorg_input, wacom tablets,...
- each user must adjust the settings by its own
- bad implementation

To use the plugin download the parts from [my xournalpp fork](https://github.com/TimoJoFischer/xournalpp/tree/wacom_plugin/plugins/Wacom_ScreenArea) and change the settings of main.lua

**Comment by ugtar** [comment]
I came here looking for something like this. More specifically, would it be possible to automatically map the wacom tablet geometry to the xournal++ window (as opposed to selecting a fixed rectangle)? That would make moving the stylus on the tablet correspond to the location on the xournal++ sheet (as opposed to the window), and would make writing like this much more intuitive

**Reply by TimoJoFischer** [reply]
@ugtar if your xournal++ windows is always at the same position with the same size, it should be do able.

In linux you can map your wacom tablet to the display with the following command:

`xsetwacom set $x MapToOutput 2020x1440+0+0`

In order to keep the proportions of the xournalpp window and the tablet the same, i aswell would cut the tablet:
`xsetwacom set $x Area 0 0 13327 9500`

Of course, you need to adapt the values to your setup. You can read more about the commands in [Commands for GraphicTablet](https://wiki.archlinux.org/title/Graphics_tablet)

**Reply by karl-tran-2002** [reply]
Relative mode (Mouse mode) is perfect for large screen

**Comment by vladQll** [comment]
and wouldn't it be the same to go to the wacom driver and in the settings adjust the area you need to work on?
![image](https://github.com/xournalpp/xournalpp/assets/28522011/3b672aea-4a44-473e-9c5c-e2551313811e)

**Reply by TimoJoFischer** [reply]
yeah but with my script, i can move the used display area (with Alt+W/A/S/D the red area is moved around)
